app/src/main/java/vn/mbm/phimp/me/MapsActivity.java

```diff
@@ -0,0 +1,107 @@
+package vn.mbm.phimp.me;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.net.Uri;
+import android.os.Bundle;
+import android.provider.MediaStore;
+import android.support.v4.app.FragmentActivity;
+import android.util.Log;
+
+import com.google.android.gms.maps.CameraUpdateFactory;
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.SupportMapFragment;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class MapsActivity extends FragmentActivity implements OnMapReadyCallback {
+
+    private GoogleMap mMap;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_maps);
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+    }
+
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        mMap = googleMap;
+
+        mMap.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
+
+            @Override
+            public boolean onMarkerClick(Marker arg0) {
+                Intent i = new Intent(MapsActivity.this, MapImage.class);
+                i.putExtra("path", arg0.getTitle());
+                Log.e("Image", arg0.getTitle());
+                startActivity(i);
+                return true;
+            }
+
+        });
+
+        ArrayList<String> listOfAllImages = getImagesPath(this);
+        int i;
+
+        for (i = 0; i < listOfAllImages.size(); i++) {
+            ExifInterface exif = null;
+            try {
+                exif = new ExifInterface(listOfAllImages.get(i));
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            float[] latLong = new float[2];
+            boolean hasLatLong = exif.getLatLong(latLong);
+            if (hasLatLong) {
+                System.out.println("Latitude: " + latLong[0]);
+                System.out.println("Longitude: " + latLong[1]);
+
+                mMap.addMarker(new MarkerOptions()
+                        .position(new LatLng(latLong[0], latLong[1]))
+                        .title(listOfAllImages.get(i)));
```

Comment:
Add the customized info screen using InfoWindowAdapter, to show image on snippet.